initrd-progs/0initrd/init

```diff
@@ -1030,42 +1044,55 @@ PUP_LAYER="$SFS_MP"
 #have basic system, now try to add optional stuff
 find_onepupdrv "$F_PART" "$F_BP_FN" "$F_DEF_FN" "f" ""
 [ "$ONE_FN" ] && FDRV="$ONE_PART,$ONE_FS,$ONE_FN"
-[ "$FDRV" ] && { LOADMSG="fdrv"; setup_onepupdrv "$FDRV" "f"; }
 
 find_onepupdrv "$Z_PART" "$Z_BP_FN" "$Z_DEF_FN" "z" ""
 [ "$ONE_FN" ] && ZDRV="$ONE_PART,$ONE_FS,$ONE_FN"
-[ "$ZDRV" ] && { LOADMSG="zdrv"; setup_onepupdrv "$ZDRV" "z"; }
 
 find_onepupdrv "$Y_PART" "$Y_BP_FN" "$Y_DEF_FN" "y" ""
 [ "$ONE_FN" ] && YDRV="$ONE_PART,$ONE_FS,$ONE_FN"
-[ "$YDRV" ] && { LOADMSG="ydrv"; setup_onepupdrv "$YDRV" "y" "p"; }
 
 find_onepupdrv "$B_PART" "$B_BP_FN" "$B_DEF_FN" "b" ""
 [ "$ONE_FN" ] && BDRV="$ONE_PART,$ONE_FS,$ONE_FN"
-[ "$BDRV" ] && { LOADMSG="bdrv"; setup_onepupdrv "$BDRV" "b" "p"; }
 
 find_onepupdrv "$A_PART" "$A_BP_FN" "$A_DEF_FN" "a" ""
 [ "$ONE_FN" ] && ADRV="$ONE_PART,$ONE_FS,$ONE_FN"
-[ "$ADRV" ] && { LOADMSG="adrv"; setup_onepupdrv "$ADRV" "a" "p"; }
 
 find_onepupdrv "" "" "$KBUILD_DEF_FN" "k" ""
 [ "$ONE_FN" ] && KBUILD="$ONE_PART,$ONE_FS,$ONE_FN"
-[ "$KBUILD" ] && { LOADMSG="kbuild"; setup_onepupdrv "$KBUILD" "k" "p"; }
 
 if [ "$UNIONFS" = 'overlay' ]; then
  find_onepupdrv "" "" "$DEVX_DEF_FN" "k" ""
  [ "$ONE_FN" ] && DEVX="$ONE_PART,$ONE_FS,$ONE_FN"
- [ "$DEVX" ] && { LOADMSG="devx"; setup_onepupdrv "$DEVX" "dev" "p"; }
 
  find_onepupdrv "" "" "$DOCX_DEF_FN" "k" ""
  [ "$ONE_FN" ] && DOCX="$ONE_PART,$ONE_FS,$ONE_FN"
- [ "$DOCX" ] && { LOADMSG="docx"; setup_onepupdrv "$DOCX" "doc" "p"; }
 
  find_onepupdrv "" "" "$NLSX_DEF_FN" "k" ""
  [ "$ONE_FN" ] && NLSX="$ONE_PART,$ONE_FS,$ONE_FN"
- [ "$NLSX" ] && { LOADMSG="nlsx"; setup_onepupdrv "$NLSX" "nls" "p"; }
 fi
 
+#if possible, copy SFSs to RAM and prioritize frequently-accessed ones
+[ "$ADRV" ] && copy_onepupdrv "$ADRV"
+[ "$YDRV" ] && copy_onepupdrv "$YDRV"
+[ "$BDRV" ] && copy_onepupdrv "$BDRV"
+[ "$DEVX" ] && copy_onepupdrv "$DEVX"
+[ "$NLSX" ] && copy_onepupdrv "$NLSX"
+[ "$DOCX" ] && copy_onepupdrv "$DOCX"
+[ "$ZDRV" ] && copy_onepupdrv "$ZDRV"
+[ "$FDRV" ] && copy_onepupdrv "$FDRV"
+[ "$KBUILD" ] && copy_onepupdrv "$KBUILD"
```

Comment:
@01micko @mavrothal Is this the right priority? adrv is top-priority, kbuild is used only rarely, zdrv and fdrv are mostly needed at boot time so they're unimporant.

I'm not sure if the order of ydrv, bdrv, devx, nlsx and docx makes sense:
- Locales on disk can slow down pretty much any application, including things like `cp`, but nlsx is very big.
- bdrv is tiny, but it may contain libraries or configuration files used by many applications.
- Many puppies are built with commonly used libraries in devx (gmp and mpfr are good examples), so if you install an application that depends on these libraries, you want devx in RAM.
- docx is relatively small but some users use it a lot: for example, I use man pages very often.
- ydrv is a wildcard, it can be big or small, depending on the user, but it probably should be in RAM because sometimes it's used to override parts of the main SFS.